### ISSUE TITLE
Julia Regex Bug?

### ISSUE THREAD

**Issue by hnkchnsk** [JuliaLang/julia#12663]
Hi all,

sadly I opened this in the wrong place: http://discuss.junolab.org/t/julia-regex-bug-not-sure-as-i-am-new-to-julia/288

In short, I am new to julia, but I think this is a bug and not me messing up program logic:

```
c=0
for i in 1:1:100000
    
	a=rand(1:10, 100)	
	b=""
	for number in a
		if number == 1
		b=b * "A"
		else b=b * "B"
		end
	
	end
	
	if ismatch(r".*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A.*A", b)
	c = c + 1
	end

end

print("Result: ")
print(c/100000)
```

I use this to estimate some statistical stuff. I had it working in other languages (even in bash ;-), but not that blazingly fast as in julia - I have it working without regex, using numbers.

Yet with the regex julia comes up with an error after some seconds:

```
ERROR: error -8
 in error at error.jl:21
 in exec at ./pcre.jl:149
 in ismatch at ./regex.jl:106
 in anonymous at no file:13
```

I checked, the BBBABBBBBA...-stuff is created as expected, c counts up to something between 0 and 75 before I hit this regex error. 

Tested on julia version 0.2.1 and 0.3.11.

Kind regards

hnk

**Comment by stevengj** [comment]
I can reproduce the problem in Julia 0.3.8, but the code runs successfully in Julia 0.4.

@malmaud, could this have been fixed by the transition to PCRE2 (#11447)?

**Comment by tkelman** [comment]
Note that due to #11577 we shouldn't upgrade release-0.3 to any of 8.34, 8.35, 8.36, 8.37 without patches for the buffer overflow. 10.20 has the fix for that but might be too large of a change to backport.

**Comment by malmaud** [comment]
Mmm, I have no special insight into what's going wrong. I don't recall any
of the pcre release notes mentioning patching something like this, but I
could have missed it.

**Comment by JeffBezanson** [comment]
Fixed in 0.4.
